### DISCUSSION TITLE
Typescript error with useTextField and useRef

### DISCUSSION THREAD

**Discussion by DanBlakeNZ** [adobe/react-spectrum#2556] [Q&A] [answered]
Hi there, 

I am trying to implement [useTextField](https://react-spectrum.adobe.com/react-aria/useTextField.html) but I am getting a TypeScript error related to the use of the 'ref' hook on the input tag. I've played around with different variations, but it just seems to just move the error around.
If I remove {...inputProps} from the input then all is fine however so I'm not sure if there is something I'm overlooking and have missed or if it is something else.

[Code Sandbox]( https://codesandbox.io/s/cocky-varahamihira-cmzxz?file=/src/TextInputField.tsx) of the issue I'm facing.

I have used the code [here](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/textfield/stories/useTextField.stories.tsx) as a base to build the above.

Any help  here would be much appreciated.

Thanks!

**Comment by bdow** [comment, marked as answer]
The `useTextField` is returning attributes for either an HTMLInputElement, or HTMLTextAreaElement. When you spread the props onto an `<input />`, you can cast it and that'd fix your TS errors:
```
<input {...(inputProps as React.InputHTMLAttributes<HTMLInputElement>)} ref={ref} />
```

**Reply by DanBlakeNZ** [reply]
Thanks @bdow, appreciate your explanation and help here. It has indeed fixed my error, thanks!
